docs/control-structures.rst

```diff
@@ -679,22 +681,43 @@ the changes in the caller will always be reverted.
     which consumed all gas available to the call.
     Exceptions that use ``require`` used to consume all gas until before the Metropolis release.
 
+.. _revert-statement:
+
 ``revert``
 ----------
 
-The ``revert`` function is another way to trigger exceptions from within other code blocks to flag an error and
-revert the current call. The function takes an optional string
-message containing details about the error that is passed back to the caller
-and it will create an ``Error(string)`` exception.
+A direct revert can be triggered using the ``revert`` statement and the ``revert`` function.
+
+The ``revert`` statement takes a custom error as direct argument without parentheses:
+
+    revert CustomError(arg1, arg2);
 
-The following example shows how to use an error string together with ``revert`` and the equivalent ``require``:
+For backards-compatibility reasons, there is also the revert function, which uses parentheses
+and accepts a string:
+
+    revert();
+    revert("description");
+
+The error data will be passed back to the caller and can be caught there.
+Using ``revert()`` causes a revert without any error data while ``revert("description")``
```

Comment:
Do we really want to allow reverts without error data?

Reply:
I mean, we already do allow them, don't we? :)

So you mean it's because we want to discourage such reverts? Well, I'd say it depends whether people do use them. If they do, I think the new syntax should have an equivalent. But I don't remember seeing many bare reverts like this in Solidity code so maybe it's ok.

Reply:
I would wait and see if we have any requests. If there are none, we can remove the revert function with 0.9.0 altogether, otherwise we add an empty revert statement.